### PR TITLE
Setup continuous integration for api

### LINKED ISSUE #13: Setup continuous integration for api

### PR DESCRIPTION
**Pull request by calebschoepp**
## Changes
- Setup continuous integration for `api`.
- Disallow lint warnings... Too strict? I kind of like the strictness to keep our codebase squeaky clean.
- Fix warnings and modify tests as needed

## TODO
- Setup continuous integration for `www`.
- Cache `npm i` in actions

Tickets have been added

Closes #13

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
     "test": "jest",
     "test:watch": "jest --watch",
     "format": "prettier --config ../.prettierrc --ignore-path .prettierignore --write .",
-    "lint": "eslint src --ext .js,.ts"
+    "format:check": "prettier --config ../.prettierrc --ignore-path .prettierignore --check .",
```
Comment (armiantos) [inline api/package.json:15]: Doesn't lint already do this (i.e. any format warnings are already done in `npm run lint`)?

**Review by armiantos — COMMENTED** [review]
One comment regarding `format:check`. Also is it normal to do build > test > lint? I usually do lint > build > test. Regarding strictness, I also prefer stricter rules so no warnings allowed is :+1:  from me

**Comment by calebschoepp** [comment]
> Also is it normal to do build > test > lint? I usually do lint > build > test.

Are you just suggesting that I should change the order of the CI jobs? That makes sense to me.

**Comment by armiantos** [comment]
Might be a good idea, I do linting first under the assumption that build may take a while and might be best to save resources if we can find early errors from linting

**Review by armiantos — APPROVED** [review]
LGTM!

**Pull request merged**